x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +101,143 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
 
+func (k Keeper) withdrawContinuousFund(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+
+	toDistributeAmount, err := k.ToDistribute.Get(ctx)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+	if toDistributeAmount.Equal(math.ZeroInt()) {
+		return sdk.Coin{}, nil
+	}
```

Comment:
if its 0 we should skip the iterate and only withdraw from the recipients allocated tokens. currently it seems we would error out and the user would not be able to withdraw their funds

Reply:
If there is no amount to distribute why would we have any amount to withdraw?

Reply:
todistirbute needs to be set to 0 after iterate. otherwise the community pool could be drained